allensdk/brain_observatory/behavior/behavior_project_cache/behavior_project_cache.py

```diff
@@ -106,6 +106,36 @@ def __init__(
         self.fetch_tries = fetch_tries
         self.logger = logging.getLogger(self.__class__.__name__)
 
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
```

Comment:
If an end-user has already previously downloaded a manifest and data files what is the mechanism for just using their local cache?

It seems like we're missing?

```
@classmethod
def from_local_cache(cls, local_cache_dir: Union[str, Path]):
```

Is this a feature we want to support at release or are we planning on adding this feature after release?

Reply:
Do you think that will happen for end users? It seems like if we release new data, we want them to use that.

This is a lower-level way to do that:
```
import allensdk.api.cloud_cache.cloud_cache as cc
from allensdk.brain_observatory.behavior.project_apis.data_io import BehaviorProjectCloudApi
   
mycache = cc.S3CloudCache(cache_dir="/home/danielk/tmp", bucket_name="single-mouse-prerelease", project_name="visual-behavior-ophys")

mycache.manifest_file_names
Out[4]: 
['visual-behavior-ophys_project_manifest_v0.1.0.json',
 'visual-behavior-ophys_project_manifest_v0.2.0.json',
 'visual-behavior-ophys_project_manifest_v0.3.0.json']

mycache.load_manifest("visual-behavior-ophys_project_manifest_v0.2.0.json")
bcapi = BehaviorProjectCloudApi(mycache)
ophys_session_table.csv: 100%|████████████████████| 3.32k/3.32k [00:00<00:00, 101kMB/s]
behavior_session_table.csv: 100%|█████████████████| 20.6k/20.6k [00:00<00:00, 306kMB/s]
ophys_experiment_table.csv: 100%|█████████████████| 13.0k/13.0k [00:00<00:00, 296kMB/s]

bc = bpc.VisualBehaviorOphysProjectCache(bcapi)
```
We can certainly make a canned version of this that would be easy. Let's discuss tomorrow what we want to offer.